cratedb_toolkit/adapter/pymongo/collection.py

```diff
@@ -124,13 +124,14 @@ def gen() -> Iterator[Mapping[str, Any]]:
                 """A generator that validates documents and handles _ids."""
                 for document in documents:
                     common.validate_is_document_type("document", document)
-                    if not isinstance(document, RawBSONDocument):
-                        if "_id" in document:
-                            identifier = ObjectId(document["_id"])
-                        else:
-                            identifier = ObjectId()
-                            document["_id"] = str(identifier)  # type: ignore[index]
-                        inserted_ids.append(identifier)
+                    if "_id" in document:
+                        identifier = ObjectId(document["_id"])
+                        if isinstance(document, RawBSONDocument):
+                            document = document.decode()
```

Comment:
This change looks a bit unrelated to the `_id` handling, why is this needed now?